cpg-core/src/main/kotlin/de/fraunhofer/aisec/cpg/TranslationManager.kt

```diff
@@ -146,6 +146,13 @@ private constructor(
 
             var sourceLocations: List<File> = ctx.config.softwareComponents[sc] ?: listOf()
 
+            sourceLocations =
```

Comment:
I think this is not enough, I think you need to move the exclusion handling down to the file walk (or after the walk), because at this point here the locations are not "expanded". For example, if I set the source location "mypackage", which has a "tests" folder in it. In this case the exclusion filter filters based on mypackage, but not on mypackage/tests which is only expanded about 10 lines down.